doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,292 @@
+	      "description": "All spec fields have matching descriptors in the CSV",
+	      "state": "fail",
+	      "suggestions": [
+		"Add a spec descriptor for size"
+	      ],
+	      "crname": "example-memcached"
+      }, 
+      {
+	      "name": "Some other test",
+	      "description": "Another test on the CSV",
+	      "state": "pass",
+	      "suggestions": [
+	      ],
+	      "crname": "example-memcached"
+      },
+    ]
+```
+
+The test image will be executed as a Pod by the scorecard with a restart policy of `never`.
+
+When executing a scorecard test, scorecard will create an environment variable in the test container that holds the name of the CR being tested, this allows the test writer a means of knowing which CR is being created by the scorecard as part of this test execution.
+
+The environment variable created in the test image container will be as follows:
+```
+CR_NAME=<some CR name>
+```
+
+Also, for each test execution, scorecard would also make the
+bundle image contents available to test images by mounting
+the bundle image contents into a shared emptyDir volume mount
+at the following location:
+
+```
+/scorecard/bundle
+```
+
+This allows tests to have access to all the bundle contents if they 
+are required for the test logic.
+
+
+For the case of executing a test, the bundle image will be passed as a scorecard command flag:
+```
+operator-sdk scorecard --bundle-image='quay.io/myorg/mytest' --selector=suite=basic
+```
+
+A command will need to be built as part of this proposal to allow
+end users the ability to pass in the scorecard test files when building
+their operator bundle images.
+
+For this, scorecard would need to support bundle images which is included as a User Story in this proposal.  
+
+#### Test Output
+
+Custom tests are expected to produce log output that conforms to the
+scorecard v1alpha2 Test Result output.  This allows scorecard to parse
+the custom test log output and aggregate the results like any other
+scorecard test.
+
+The scorecard expected output would be in JSON, and would match the following definition:  https://github.com/operator-framework/operator-sdk/blob/master/pkg/apis/scorecard/v1alpha2/types.go#L36
+
+The scorecard v1alpha2 Test Result output contains information such as
+pass or fail status and suggestions/errors produced by the test that
+the end user would want to see in the aggregated scorecard results.
+
+
+### Custom Test Development Model
+
+The developer of a custom test would have a workflow similar to:
+ 
+ * write a custom test binary in the language of their choice, sample custom test within the SDK repository would be available as a starting point
+ * test the custom test binary locally (out-of-cluster) to make sure the output is produced that matches v1alpha2 ScorecardTestResult format
+ * create a Pod manifest to run their custom test image, copying a sample Pod manifest from the scorecard examples in the SDK repository
```

Comment:
It would be nice to have a script/set of flags in `scorecard` to generate a basic Pod manifest for a test, with some `# Add <thing> here` comments in various fields that we can't fill in via CLI. Not necessarily something that needs codifying in this document, just a thought for a followup.

Reply:
good idea, I was thinking either good examples people could copy but I see value in some scaffolding to generate a test based on a given operator's configuration/naming.